File: 01 - Dart/helloworld.dart
// Día 1: Dart
// Clase en vídeo: https://www.twitch.tv/videos/1786404728?t=00h12m34s

void main() {
  /*
   * Esto es un comentario
   */

  // Hola mundo

  print("Hola, Dart!");

  // Variables

  var myString = "Esto es una cadena de texto";
  myString = "Aquí cambio el valor de la cadena de texto";
  // helloDart = 6; Error
  print(myString);

  String myString2 = "Esto es otra cadena de texto";
  print(myString2);

  var myInt = 7;
  myInt = myInt + 4;
  print(myInt);
  print(myInt - 1);
  print(myInt);

  int myInt2 = 5;
  print(myInt2);

  var myDouble = 6.5;
  print(myDouble);

  myDouble = 6;
  print(myDouble);

  double myDouble2 = 123.56;
  print(myDouble2);

  var myBool = false;
  myBool = true;
  print(myBool);

  // Constantes

  final myFinal = "Mi propiedad final";
  // myFinal = "Mi nueva propiedad final"; Error

  final myFinalInt = myInt;
  print(myFinalInt);

  const myConst = "Mi propiedad constante";
  // myConst = "Mi nueva propiedad constante"; Error
  // const myConstInt = myInt; Error

  // Control de flujo

  if (myInt == 10 && myString == "Hola") {
    print("El valor es 10");
  } else if (myInt == 11 || myString == "Hola") {
    print("El valor es 11");
  } else {
    print("El valor no es 10 ni 11");
  }

  // Listas

  myFunction();
  var myReturn = myFunctionWithReturn();
  print(myReturn);

  var myList = ["Brais", "Moure", "@mouredev", "Brais"];
  print(myList);
  print(myList[0]);

  // Sets

  var mySet = {"Brais", "Moure", "@mouredev", "Brais"};
  print(mySet);

  // Mapas

  var myMap = {"Brais": 35, "Genemix": 23};
  print(myMap);
  print(myMap["Brais"]);

  var myMap2 = Map<String, int>();
  myMap2["Frg152"] = 22;
  myMap2["Zhamudio"] = 20;

  // Bucles

  for (final value in myList) {
    print(value);
  }

  var myCounter = 0;

  while (myCounter <= myInt) {
    print(myCounter);
    myCounter++;
  }

  // Clases

  var myClass = MyClass("Brais", 35);
  print(myClass.name);
  print(myClass.age);

  // Opcionales

  String? myOptionalString;
  print(myOptionalString);

  myOptionalString = "Mi valor no nulo";
  print(myOptionalString);
  myOptionalString = null;
  print(myOptionalString);

  // Enum

  print(MyEnum.dart);
}

// Funciones

void myFunction() {
  print("Esto es una función");
}

String myFunctionWithReturn() {
  return "Esto es una función con retorno";
}

// Clases

class MyClass {
  String name;
  int age;

  MyClass(this.name, this.age) {}
}

// Enum

enum MyEnum { dart, python, swift, java, kotlin }

File: 06 - JavaScript/hello_world.js
// Día 6: JavaScript
// Clase en vídeo: https://youtu.be/6cnFl9aHD5Y

/*
Esto es un comentario
*/

// Hola mundo

console.log("Hola, JavaScript!");

// Variables

var myString = "Esto es una cadena de texto"
console.log(myString)

let myString2 = "Esto es una cadena de texto"
myString2 = "Aquí cambio el valor de la cadena de texto"
console.log(typeof myString2)
myString2 = 6
console.log(myString2)
console.log(typeof myString2)

let myNumber = 7
myNumber = myNumber + 4
console.log(myNumber)
console.log(myNumber - 1)
console.log(myNumber)

console.log(myString + " " + myNumber)

let myNumber2 = 6.5
console.log(myNumber2)
console.log(typeof myNumber2)

let myBool = false
myBool = true
console.log(myBool)
console.log(typeof myBool)

myBool = null
console.log(myBool + myNumber2)

myBool = undefined
console.log(myBool)

let myUndefined
console.log(myUndefined)

// Constantes

const MY_CONST = "Mi propiedad constante"
// MY_CONST = "Otro valor" Error
console.log(MY_CONST)

// Control de flujo

if (myNumber == 10 && myString == "Hola") {
    console.log("El valor es 10")
} else if (myNumber == 11 || myString == "Hola") {
    console.log("El valor es 11")
} else {
    console.log("El valor no es 10")
}

myBool = null

if (myBool) {
    console.log("null")
}

// Funciones

function myFunction() {
    return "Mi función"
}

console.log(myFunction())

// Listas

let myList = ["Brais", "Moure", "@mouredev", 36]
console.log(myList)
console.log(myList[0])

// Set

let mySet = new Set(["Brais", "Moure", "@mouredev", 36, "Brais"])
mySet.add("Moure")
console.log(mySet)

// Mapas

let myMap = new Map([["Brais", 36], ["ztiake", 37], ["zPato04", 18]])
myMap.set("Niku1525", 20)
console.log(myMap)
console.log(myMap.get("Brais"))

// Bucles

for (const value of myList) {
    console.log(value)
}

let myCounter = 0

while (myCounter < myList.length) {
    console.log(myList[myCounter])
    myCounter++
}

// Clases

class MyClass {
    constructor(name, age) {
        this.name = name
        this.age = age
    }
}

let myClass = new MyClass("Brais", 36)
console.log(myClass)
console.log(myClass.name)

// Enum

const MyEnum = {
    DART: "dart",
    PYTHON: "python",
    SWIFT: "swift",
    JAVA: "java",
    KOTLIN: "kotlin",
    JAVASCRIPT: "javascript"
}

const myEnum = MyEnum.JAVASCRIPT
console.log(myEnum)

